File: Android/CurrencyConverter/app/src/main/java/com/ams/currencyconverter/common/exceptionsAndThrowables/APIError.kt
package com.ams.currencyconverter.common.exceptionsAndThrowables

@Suppress("unused")
data class APIError(val success: Boolean, val error: Map<String, Any>): Throwable() {

    fun getErrorCode(): Int = error["code"] as Int

    fun getErrorMessage(): String = error["info"] as String
}

File: iOS/CurrencyConverter/CurrencyConverter/Common/Data/Models/APIError.swift
//
//  APIError.swift
//  CurrencyConverter
//
//  Created by Ahmad Mahmoud on 14/12/2020.
//

struct APIError: Error {
    
    let success: Bool
    let error: FixerAPIErrorDetails
    
    init(success: Bool, error: FixerAPIErrorDetails) {
        self.success = success
        self.error = error
    }
    
    func getErrorCode() -> Int {
        return error.code
    }

    func getErrorMessage() -> String {
        return error.info
    }
}
